cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -235,24 +234,20 @@ type annotator struct {
 }
 
 func parseNodeURL(nodeURL string) (project, zone, instance string, err error) {
-	u, err := url.Parse(nodeURL)
-	if err != nil {
-		return "", "", "", fmt.Errorf("failed to parse %q: %v", nodeURL, err)
+	// We only expect to handle strings that look like:
+	// gce://project/zone/instance. Splitting by "/", the parts should look
+	// like: ["gce:", "", "project", "zone", "instance"]
+	parts := strings.Split(nodeURL, "/")
+	if len(parts) != 5 {
+		return "", "", "", fmt.Errorf("failed to parse %q: expected a three part path", nodeURL)
 	}
-	if u.Scheme != "gce" {
+	if parts[0] != "gce:" {
 		return "", "", "", fmt.Errorf("instance %q doesn't run on gce", nodeURL)
 	}
```

Comment:
use `strings.TrimPrefix` to remove `gce://` before spliting

Reply:
Done.